domain/miningmanager/mempool/model/transaction.go

```diff
@@ -0,0 +1,9 @@
+package model
+
+import "github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+
+// Transaction represents a generic transaction either in the mempool's main TransactionPool or OrphanPool
+type Transaction interface {
```

Comment:
Already have MempoolTransaction for something else.
I can rename it to GenericTransaction or something, but first, let me try argue it's not needed.

This is in module mempool/model, which is not accessed from anywhere except mempool package. Therefore it should be fine.
(Of course, it could be nice if I could have enforced that nobody accesses it outside mempool, but Golang doesn't let me do that)